app/jobs/export_structural_job.rb

```diff
@@ -26,32 +22,24 @@ def perform(bulk_action_id, _params)
   private
 
   # @return [Array<Array>] Returns an array of rows for the object
-  def item_to_rows(druid, log_buffer)
+  def item_to_rows(item, success, failure)
     result = []
-    log_buffer.puts("#{Time.current} #{self.class}: Loading #{druid}")
-    item = Dor::Services::Client.object(druid).find
     if !item.dro? || item.structural.contains.empty?
-      log_failure("Object #{druid} has no structural metadata to export", log_buffer)
+      failure.call('No structural metadata to export')
       return []
     end
 
-    log_buffer.puts("#{Time.current} #{self.class}: Exporting structural metadata for #{druid}")
     StructureSerializer.new(item.structural).rows do |row|
       result << row
     end
-    bulk_action.increment(:druid_count_success).save
+    success.call('Exported structural metadata')
     result
   rescue StandardError => e
-    log_buffer.puts("#{Time.current} #{self.class}: Unexpected error exporting structural metadata for #{druid}: #{e}")
-    bulk_action.increment(:druid_count_fail).save
+    failure.call("Unexpected error exporting structural metadata: #{e.message}")
+    Honeybadger.notify(e)
```

Comment:
Does this need a context? Does it need to be rescued at all or is it rescued by `with_items`?